plugin/builtin/input/file/file.go

```diff
@@ -134,7 +116,37 @@ func (c FileInputConfig) Build(context plugin.BuildContext) (plugin.Plugin, erro
 	return plugin, nil
 }
 
-type FileInput struct {
+// getSplitFunc will return the split function associated the configured mode.
+func (c InputConfig) getSplitFunc() (bufio.SplitFunc, error) {
+	var splitFunc bufio.SplitFunc
+	if c.Multiline == nil {
+		splitFunc = NewNewlineSplitFunc()
+	} else {
+		definedLineEndPattern := c.Multiline.LineEndPattern != ""
+		definedLineStartPattern := c.Multiline.LineStartPattern != ""
+
+		switch {
+		case definedLineEndPattern == definedLineStartPattern:
+			return nil, fmt.Errorf("if multiline is configured, exactly one of line_start_pattern or line_end_pattern must be set")
+		case definedLineEndPattern:
+			re, err := regexp.Compile(c.Multiline.LineEndPattern)
+			if err != nil {
+				return nil, fmt.Errorf("compile line end regex: %s", err)
+			}
+			splitFunc = NewLineEndSplitFunc(re)
+		case definedLineStartPattern:
+			re, err := regexp.Compile(c.Multiline.LineStartPattern)
+			if err != nil {
+				return nil, fmt.Errorf("compile line start regex: %s", err)
+			}
+			splitFunc = NewLineStartSplitFunc(re)
+		}
+	}
+	return splitFunc, nil
+}
+
+// InputPlugin is a plugin that monitors files for entries
+type InputPlugin struct {
```

Comment:
I know it's probably complaining about stuttering, but I find this change a step backwards, making it harder to figure out what's going on at a glance.

Reply:
The other option we have is to rename it `Plugin` and `Config`. That way, when imported, they'd read as `file.Plugin` and `file.Config`.